tests/acceptance/test_embedded_study.py

```diff
@@ -13,3 +17,39 @@ def test_generate_dataset(study, database):
 
     assert len(results) == 1
     assert results[0]["year"] == "1943"
+
+
+def test_validate_dummy_data_happy_path(tmp_path):
+    dummy_data = "patient_id,year\n1,1971\n2,1992"
+    dataset_def_path = tmp_path / "dataset.py"
+    dummy_data_path = tmp_path / "dummy_data.csv"
+    dataset_path = tmp_path / "dataset.csv"
+    dataset_def_path.write_text(trivial_dataset_definition)
+    dummy_data_path.write_text(dummy_data)
+    _generate_dataset(dataset_def_path, dummy_data_path, dataset_path)
+    assert dataset_path.read_text() == dummy_data
+
+
+def test_validate_dummy_data_error_path(tmp_path):
+    dummy_data = "patient_id,year\n1,1971\n2,foo"
+    dataset_def_path = tmp_path / "dataset.py"
+    dummy_data_path = tmp_path / "dummy_data.csv"
+    dataset_path = tmp_path / "dataset.csv"
+    dataset_def_path.write_text(trivial_dataset_definition)
+    dummy_data_path.write_text(dummy_data)
+    with pytest.raises(ValidationError, match="Invalid int"):
+        _generate_dataset(dataset_def_path, dummy_data_path, dataset_path)
```

Comment:
I'm not sure what we gain by testing this error path here. Is there anything this covers that isn't covered by the unit tests?

Reply:
I added it because it confirms that the entrypoint does actually call the validation code. If we didn't have this test we could delete those calls and just always copy the dummy data file over and all the tests would still pass, which felt wrong to me.

Reply:
I think that I'd tackle that by refactoring the code so there is a single function which handles dummy data functionality -- both validating and copying the file -- and unit test that. I don't feel strongly about this, though.

Reply:
Maybe? Really up to you.